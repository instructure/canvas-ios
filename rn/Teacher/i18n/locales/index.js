//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

// needs to be commonjs so eslint-plugin-format-message can read it
module.exports = {
  ar: require('./ar.json'),
  ca: require('./ca.json'),
  cy: require('./cy.json'),
  da: require('./da.json'),
  'da-instk12': require('./da-instk12.json'),
  de: require('./de.json'),
  'en-AU': require('./en-AU.json'),
  'en-AU-unimelb': require('./en-AU-unimelb.json'),
  'en-CY': require('./en-CY.json'),
  'en-GB': require('./en-GB.json'),
  'en-GB-instukhe': require('./en-GB-instukhe.json'),
  en: require('./en.json'),
  es: require('./es.json'),
  fi: require('./fi.json'),
  fr: require('./fr.json'),
  'fr-CA': require('./fr-CA.json'),
  ht: require('./ht.json'),
  is: require('./is.json'),
  ja: require('./ja.json'),
  mi: require('./mi.json'),
  nb: require('./nb.json'),
  'nb-instk12': require('./nb-instk12.json'),
  nl: require('./nl.json'),
  pl: require('./pl.json'),
  pt: require('./pt.json'),
  'pt-BR': require('./pt-BR.json'),
  ru: require('./ru.json'),
  sl: require('./sl.json'),
  sv: require('./sv.json'),
  'sv-instk12': require('./sv-instk12.json'),
  zh: require('./zh.json'),
  'zh-HK': require('./zh-HK.json'),
}
