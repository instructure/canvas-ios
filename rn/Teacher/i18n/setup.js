//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import {
  NativeModules,
} from 'react-native'
import i18n from 'format-message'
import translations from './locales/index'

// Some crashing happening because a locale is not returned, so I've made this an optional
// and if we can't get a locale, default to english
export function sanitizeLocale (locale: ?string): string {
  if (!locale) return 'en'
  // Found some crash reports with this causing issues.
  // Apparently apple uses that locale sometimes and it just means english.
  if (locale === 'en-US_POSIX') return 'en'
  if (locale.indexOf('@') !== -1) {
    const index = locale.indexOf('@')
    locale = locale.substr(0, index)
  }
  return locale.replace('_', '-')
}

let currentLocale: string = ''
export function getLocale (): string {
  if (!currentLocale) {
    console.warn('You have accessed the locale before it has been set. Are you positive you want to do this?')
    return NativeModules.SettingsManager.settings.AppleLocale
  }

  return currentLocale
}

export default function (locale: ?string): void {
  const sanitizedLocale = sanitizeLocale(locale)
  currentLocale = sanitizedLocale
  i18n.setup({
    // generateId underscored_crc32 done via babel transform
    locale: sanitizedLocale,
    translations,
    missingTranslation: 'ignore',
    formats: {
      number: {
        default: {
          style: 'decimal',
          minimumFractionDigits: 0,
          maximumFractionDigits: 2,
        },
        percent: {
          style: 'percent',
          minimumFractionDigits: 0,
          maximumFractionDigits: 2,
        },
        grade: {
          style: 'decimal',
          minimumFractionDigits: 0,
          maximumFractionDigits: 2,
        },
      },
      date: {
        'MMMM d': { month: 'long', day: 'numeric' },
        'MMM d': { month: 'short', day: 'numeric' },
        'M/d/yyyy': { day: 'numeric', month: 'numeric', year: 'numeric' },
        'MMM d, YYYY': { day: 'numeric', month: 'short', year: 'numeric' },
      },
    },
  })
}
