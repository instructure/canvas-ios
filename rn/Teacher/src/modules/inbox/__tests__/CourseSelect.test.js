//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React from 'react'
import {
  CourseSelect,
  mapStateToProps,
  shouldRefresh,
  doRefresh,
  isRefreshing } from '../CourseSelect'
import renderer from 'react-test-renderer'
import explore from '../../../../test/helpers/explore'
import i18n from 'format-message'

jest.mock('react-native/Libraries/Components/Touchable/TouchableHighlight', () => 'TouchableHighlight')

let template = {
  ...require('../../../__templates__/course'),
  ...require('../../../redux/__templates__/app-state'),
  ...require('../../../__templates__/helm'),
}

let c1 = template.course({
  is_favorite: true,
  id: '1',
})

let c2 = template.course({
  is_favorite: false,
  id: '2',
})

let c3 = template.course({
  is_favorite: false,
  id: '3',
})

let c4 = template.course({ id: '4', workflow_state: 'completed' })

let defaultProps = {
  navigator: template.navigator({
    dismiss: jest.fn(),
  }),
  onSelect: jest.fn(),
  pending: false,
  sections: [
    {
      key: 0,
      title: 'favorites',
      data: [c1],
    },
    {
      key: 1,
      title: 'courses',
      data: [c2, c3],
    },
  ],
}

describe('CourseSelect', () => {
  beforeEach(() => jest.clearAllMocks())

  it('renders', () => {
    let tree = renderer.create(
      <CourseSelect {...defaultProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders and then selects a course and then dismisses', () => {
    const component = renderer.create(
      <CourseSelect {...defaultProps} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    const courseRow = explore(tree).selectByID(`inbox.course-select.course-${c1.id}`) || {}
    courseRow.props.onPress()
    expect(defaultProps.onSelect).toHaveBeenCalled()
  })

  it('mapStateToProps', () => {
    const appState = template.appState({
      entities: {
        courses: {
          [c1.id]: {
            course: c1,
          },
          [c2.id]: {
            course: c2,
          },
          [c3.id]: {
            course: c3,
          },
          [c4.id]: {
            course: c4,
          },
        },
      },
    })

    let result = mapStateToProps(appState)
    expect(result).toMatchObject({
      sections: [
        {
          key: 0,
          title: i18n('Favorite Courses'),
          data: [c1],
        },
        {
          key: 1,
          title: i18n('Courses'),
          data: [c2, c3],
        },
      ],
    })
  })

  it('refresh functions', () => {
    const props = {
      refreshCourses: jest.fn(),
    }

    expect(shouldRefresh()).toEqual(true)
    doRefresh(props)
    expect(props.refreshCourses).toHaveBeenCalled()
    expect(isRefreshing({ pending: true })).toEqual(true)
  })
})
