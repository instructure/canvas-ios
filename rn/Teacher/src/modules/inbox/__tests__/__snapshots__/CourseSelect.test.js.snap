// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSelect renders 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <list>
    <section>
      <View
        accessibilityTraits="header"
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "flex": 1,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 8,
              "paddingVertical": 4,
            },
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontFamily": "Lato-Regular",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#556572",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                undefined,
              ],
            ]
          }
        >
          favorites
        </Text>
      </View>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </section>
    <section>
      <View
        accessibilityTraits="header"
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "flex": 1,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 8,
              "paddingVertical": 4,
            },
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontFamily": "Lato-Regular",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#556572",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                undefined,
              ],
            ]
          }
        >
          courses
        </Text>
      </View>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-2"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "checkSolid",
                    }
                  }
                  style={
                    Object {
                      "height": 20,
                      "marginLeft": 16,
                      "resizeMode": "contain",
                      "tintColor": "#008EE2",
                      "width": 20,
                    }
                  }
                  testID="inbox.course-select.course-2.checkmark"
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-3"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </section>
  </list>
</RCTSafeAreaView>
`;

exports[`CourseSelect renders and then selects a course and then dismisses 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <list>
    <section>
      <View
        accessibilityTraits="header"
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "flex": 1,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 8,
              "paddingVertical": 4,
            },
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontFamily": "Lato-Regular",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#556572",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                undefined,
              ],
            ]
          }
        >
          favorites
        </Text>
      </View>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </section>
    <section>
      <View
        accessibilityTraits="header"
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "flex": 1,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 8,
              "paddingVertical": 4,
            },
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontFamily": "Lato-Regular",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#556572",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                undefined,
              ],
            ]
          }
        >
          courses
        </Text>
      </View>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-2"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "checkSolid",
                    }
                  }
                  style={
                    Object {
                      "height": 20,
                      "marginLeft": 16,
                      "resizeMode": "contain",
                      "tintColor": "#008EE2",
                      "width": 20,
                    }
                  }
                  testID="inbox.course-select.course-2.checkmark"
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
      <item>
        <TouchableHighlight
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-3"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontFamily": "Lato-Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </section>
  </list>
</RCTSafeAreaView>
`;
